lib/ts/supertokens.ts

```diff
@@ -99,21 +99,44 @@ export default class SuperTokens {
         this.isInServerlessEnv = config.isInServerlessEnv === undefined ? false : config.isInServerlessEnv;
 
         let multitenancyFound = false;
+        let totpFound = false;
+        let userMetadataFound = false;
+        let multiFactorAuthFound = false;
+
         // Multitenancy recipe is an always initialized recipe and needs to be imported this way
         // so that there is no circular dependency. Otherwise there would be cyclic dependency
         // between `supertokens.ts` -> `recipeModule.ts` -> `multitenancy/recipe.ts`
         let MultitenancyRecipe = require("./recipe/multitenancy/recipe").default;
+        let UserMetadataRecipe = require("./recipe/usermetadata/recipe").default;
+        let MultiFactorAuthRecipe = require("./recipe/multifactorauth/recipe").default;
+        let TotpRecipe = require("./recipe/totp/recipe").default;
+
         this.recipeModules = config.recipeList.map((func) => {
             const recipeModule = func(this.appInfo, this.isInServerlessEnv);
             if (recipeModule.getRecipeId() === MultitenancyRecipe.RECIPE_ID) {
                 multitenancyFound = true;
+            } else if (recipeModule.getRecipeId() === UserMetadataRecipe.RECIPE_ID) {
+                userMetadataFound = true;
+            } else if (recipeModule.getRecipeId() === MultiFactorAuthRecipe.RECIPE_ID) {
+                multiFactorAuthFound = true;
+            } else if (recipeModule.getRecipeId() === TotpRecipe.RECIPE_ID) {
+                totpFound = true;
             }
             return recipeModule;
         });
 
         if (!multitenancyFound) {
             this.recipeModules.push(MultitenancyRecipe.init()(this.appInfo, this.isInServerlessEnv));
         }
+        if (totpFound && !multiFactorAuthFound) {
+            // we want MFA to be enabled if totp is enabled
+            this.recipeModules.push(MultiFactorAuthRecipe.init()(this.appInfo, this.isInServerlessEnv));
+            multiFactorAuthFound = true;
+        }
```

Comment:
why are we initing metadata in case of totp? It's supposed to only be for mfa right?

Reply:
Initializing totp auto inits MFA, which inturn auto inits user metadata